boilerplate/app/components/Toggle.tsx

```diff
@@ -0,0 +1,493 @@
+
+    return { backgroundColor }
+  }, [status, disabled, on])
+
+  const $animatedSwitchKnob = useAnimatedStyle(() => {
+    const offBackgroundColor = (function () {
+      if ($switchContainerStyleOverride?.backgroundColor) {
+        return $switchContainerStyleOverride.backgroundColor
+      }
+      if (disabled) return colors.palette.neutral600
+      if (status === "error") return colors.error
+      return colors.palette.secondary500
+    })()
+
+    const onBackgroundColor = (function () {
+      if ($switchKnobStyleOverride?.backgroundColor) return $switchKnobStyleOverride.backgroundColor
+      if (disabled) return colors.palette.neutral600
+      return colors.palette.neutral100
+    })()
+
+    const backgroundColor = withTiming(on ? onBackgroundColor : offBackgroundColor)
+    const start = withTiming(on ? "100%" : "0%")
+    const marginStart = withTiming(on ? -renderedKnobWidth : 0)
+
+    return { backgroundColor, start, marginStart }
+  }, [status, disabled, on])
+
+  const $switchContainerStyle = [
+    $switchContainer,
+    $animatedSwitchContainer,
+    $switchContainerStyleOverride,
+  ]
+  const $switchKnobStyle = [$switchKnob, $animatedSwitchKnob, $switchKnobStyleOverride]
+
+  return (
+    <Animated.View style={$switchContainerStyle}>
+      <SwitchAccessibilityLabel {...props} role="on" />
+      <SwitchAccessibilityLabel {...props} role="off" />
+
+      <Animated.View
+        style={$switchKnobStyle}
+        onLayout={(e) => setRenderedKnobWidth(Math.floor(e.nativeEvent.layout.width))}
+      />
+    </Animated.View>
+  )
+}
+
+function SwitchAccessibilityLabel(props: ToggleInputProps & { role: "on" | "off" }) {
+  const { on, disabled, status, switchAccessibilityMode, role, innerStyle, detailStyle } = props
+
+  if (!switchAccessibilityMode) return null
+
+  const shouldLabelBeVisible = (on && role === "on") || (!on && role === "off")
+
+  const $switchAccessibilityStyle = [
+    $switchAccessibility,
+    role === "off" && { end: "10%" },
+    role === "on" && { left: "10%" },
+  ]
+
+  const color = (function () {
+    if (disabled) return colors.palette.neutral600
+    if (status === "error" && !on) return colors.error
+    if (!on) return innerStyle?.backgroundColor || colors.palette.secondary500
+    return detailStyle?.backgroundColor || colors.palette.neutral100
+  })()
+
+  return (
+    <View style={$switchAccessibilityStyle}>
+      {switchAccessibilityMode === "text" && shouldLabelBeVisible && (
+        <Text
+          size="xs"
+          style={{ color }}
+          tx={role === "off" ? "accessibility.switchOff" : "accessibility.switchOn"}
+        />
+      )}
+
+      {switchAccessibilityMode === "icon" && shouldLabelBeVisible && (
+        <Image
+          style={[$switchAccessibilityIcon, { tintColor: color }]}
+          source={role === "off" ? iconRegistry.hidden : iconRegistry.view}
+        />
+      )}
+    </View>
+  )
+}
```

Comment:
This component provides more visual markers for the state of the switch. In "text" mode, we output `| or ⭘`. In "icon" mode, we have a visible/hidden icons.